com.sap.cloud.lm.sl.cf.process/src/main/java/com/sap/cloud/lm/sl/cf/process/steps/ProcessStepHelper.java

```diff
@@ -29,34 +29,30 @@
     private static final String CORRELATION_ID = "correlationId";
 
     private ProgressMessageService progressMessageService;
-    private ProcessLoggerProviderFactory processLoggerProviderFactory;
     private TaskIndexProvider taskIndexProvider;
+    private ProcessLoggerHandler processLoggerHandler;
+    private StepLogger stepLogger;
 
     private ProcessEngineConfiguration processEngineConfiguration;
 
     String taskId;
     String taskIndex;
 
-    public ProcessStepHelper(ProgressMessageService progressMessageService, ProcessLoggerProviderFactory processLoggerProviderFactory,
-        TaskIndexProvider stepIndexProvider, ProcessEngineConfiguration processEngineConfigurationSupplier) {
+    public ProcessStepHelper(ProgressMessageService progressMessageService,
+        StepLogger stepLogger, TaskIndexProvider stepIndexProvider, ProcessLoggerHandler processLoggerHandler,
+        ProcessEngineConfiguration processEngineConfigurationSupplier) {
         this.progressMessageService = progressMessageService;
-        this.processLoggerProviderFactory = processLoggerProviderFactory;
+        this.stepLogger = stepLogger;
         this.taskIndexProvider = stepIndexProvider;
+        this.processLoggerHandler = processLoggerHandler;
         this.processEngineConfiguration = processEngineConfigurationSupplier;
     }
 
     protected void postExecuteStep(DelegateExecution context, StepPhase state) {
-        logDebug(context, MessageFormat.format(Messages.STEP_FINISHED, context.getCurrentFlowElement()
+        logDebug(MessageFormat.format(Messages.STEP_FINISHED, context.getCurrentFlowElement()
             .getName()));
 
-        processLoggerProviderFactory.removeAll();
-        // Write the log messages:
-        try {
-            processLoggerProviderFactory.append(context);
-        } catch (IOException | FileStorageException e) {
-            LOGGER.warn(MessageFormat.format(Messages.COULD_NOT_PERSIST_LOGS_FILE, e.getMessage()), e);
-        }
-
+        processLoggerHandler.appendProcessLog(context);
```

Comment:
Why are the loggers not removed? Are they kept in the cache forever? This would eventually lead to a memory leak.